src/general/StringUtils.cs

```diff
@@ -187,4 +189,40 @@ public static string Unescape(this string input)
 
         return result.ToString();
     }
+
+    /// <summary>
+    ///   Convert a value in a string with up to 3 digits.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     If greater than 100, it should use F0, otherwise it should use F1.
+    ///     Above 1000 it should use the number formatting which uses K for
+    ///     thousands, M for millions etc. It can handle negative numbers
+    ///   </para>
+    /// </remarks>
+    /// <param name="value">The value to format</param>
+    /// <returns>The formatted string</returns>
+    public static string ThreeDigitFormat(double value)
+    {
+        double absoluteValue = Math.Abs(value);
+        if (absoluteValue >= 1000)
```

Comment:
I think this should use the same approach as in `FormatNumber` i.e. check if the `value is >= 1000 or <= -1000`